nms/app/packages/magmalte/app/views/equipment/GatewayDetailStatus.js

```diff
@@ -14,21 +14,39 @@
  * @format
  */
 import type {DataRows} from '../../components/DataGrid';
-import type {lte_gateway} from '@fbcnms/magma-api';
 
 import DataGrid from '../../components/DataGrid';
+import GatewayContext from '../../components/context/GatewayContext';
 import LoadingFiller from '@fbcnms/ui/components/LoadingFiller';
 import MagmaV1API from '@fbcnms/magma-api/client/WebClient';
 import React from 'react';
 import isGatewayHealthy, {DynamicServices} from '../../components/GatewayUtils';
 import nullthrows from '@fbcnms/util/nullthrows';
 import useMagmaAPI from '@fbcnms/ui/magma/useMagmaAPI';
 
+import {
+  REFRESH_INTERVAL,
+  useRefreshingContext,
+} from '../../components/context/RefreshContext';
+import {useEnqueueSnackbar} from '@fbcnms/ui/hooks/useSnackbar';
 import {useRouter} from '@fbcnms/ui/hooks';
 
-export default function GatewayDetailStatus({gwInfo}: {gwInfo: lte_gateway}) {
+export default function GatewayDetailStatus() {
   const {match} = useRouter();
   const networkId: string = nullthrows(match.params.networkId);
+  const gatewayId: string = nullthrows(match.params.gatewayId);
+  const enqueueSnackbar = useEnqueueSnackbar();
+  // Auto refresh gateways every 30 seconds
+  const state = useRefreshingContext({
+    context: GatewayContext,
+    networkId: networkId,
+    type: 'gateway',
+    interval: REFRESH_INTERVAL,
+    id: gatewayId,
+    enqueueSnackbar: enqueueSnackbar,
```

Comment:
in case of an error, let's skip printing every 30 seconds. Retain the attribute in the prop, but you can remove it here.